### PR TITLE
Fix linting rule post operation

### PR DESCRIPTION
**Pull request by AlitzelMendez**
issue: https://github.com/Azure/typespec-azure/issues/1672

### CODE REVIEW

File: .chronus/changes/fix-linting-rule-post-operation-2024-9-15-15-21-46.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+changeKind: fix
+packages:
+  - "@azure-tools/typespec-azure-resource-manager"
+---
+
+Fix linting rule to ensure that the linting rule triggers for @get operations on ArmProviderAction
```
Comment (timotheeguerin) [inline .chronus/changes/fix-linting-rule-post-operation-2024-9-15-15-21-46.md:7]:
```suggestion
Fix linting rule to ensure that the linting rule triggers for `@get` operations on ArmProviderAction
```

mark sure to escape decorators or they will tag user with that name in the changelog

Comment (timotheeguerin) [reply .chronus/changes/fix-linting-rule-post-operation-2024-9-15-15-21-46.md:7]: also message would be good to include the rule name, I changed it but feel free to update to the more accurate description of this fix

File: packages/typespec-azure-resource-manager/src/rules/arm-resource-invalid-action-verb.ts [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { Operation, createRule } from "@typespec/compiler";
 import { getOperationVerb } from "@typespec/http";
 
 import { getActionDetails } from "@typespec/rest";
-import { isInternalTypeSpec, isSourceOperationResourceManagerInternal } from "./utils.js";
+import { isInternalTypeSpec } from "./utils.js";
 
 export const invalidActionVerbRule = createRule({
   name: "arm-resource-invalid-action-verb",
```
Comment (timotheeguerin) [inline packages/typespec-azure-resource-manager/src/rules/arm-resource-invalid-action-verb.ts:8]: while you are here would you mind adding teh documentationUrl and doc page for this rule(check `no-enum.ts` for example)

File: packages/typespec-azure-resource-manager/src/rules/arm-resource-invalid-action-verb.ts [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { Operation, createRule } from "@typespec/compiler";
 import { getOperationVerb } from "@typespec/http";
 
 import { getActionDetails } from "@typespec/rest";
-import { isInternalTypeSpec, isSourceOperationResourceManagerInternal } from "./utils.js";
+import { isInternalTypeSpec } from "./utils.js";
 
 export const invalidActionVerbRule = createRule({
```
Comment (timotheeguerin) [inline packages/typespec-azure-resource-manager/src/rules/arm-resource-invalid-action-verb.ts:7]: also while we are here lets rename this to match exactly the name of the rule camelCase with `Rule` suffix`

```
armResourceInvalidActionVerbRule
```

File: packages/typespec-azure-resource-manager/test/rules/operation-usage.test.ts
Comment (timotheeguerin) [inline packages/typespec-azure-resource-manager/test/rules/operation-usage.test.ts:1]: and this test file to match the name of the rule too

File: packages/typespec-azure-resource-manager/test/rules/operation-usage.test.ts [hunk 4]
```diff
@@ -87,39 +87,6 @@ describe("typespec-azure-resource-manager: detect non-post actions", () => {
         message: "Actions must be HTTP Post operations.",
       });
   });
-
-  it("Allows post actions for authorized provider actions", async () => {
```
Comment (timotheeguerin) [inline packages/typespec-azure-resource-manager/test/rules/operation-usage.test.ts:91]: would maybe be good to keep this test now that you have it but check the opposite(still emit a warning there)

Comment (AlitzelMendez) [reply packages/typespec-azure-resource-manager/test/rules/operation-usage.test.ts:91]: I just saw this! let me add this :)

File: docs/libraries/azure-resource-manager/rules/arm-resource-invalid-action-verb.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+```text title=- Full name-
+@azure-tools/typespec-azure-resource-manager/arm-resource-invalid-action-verb
+```
+
+For ARM http operations, the action verb must be `@post`. Any other action verb is flagged as incorrect.
+
+#### ❌ Incorrect
+
+```tsp
+    @get getAction is ArmProviderActionAsync<
```
Comment (timotheeguerin) [inline docs/libraries/azure-resource-manager/rules/arm-resource-invalid-action-verb.md:14]:
```suggestion
    @get op getAction is ArmProviderActionAsync<
```

File: docs/libraries/azure-resource-manager/rules/arm-resource-invalid-action-verb.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+      },
+      ArmCombinedLroHeaders,
+      SubscriptionActionScope
+    >;
+```
+
+#### ✅ Correct
+
+```tsp
+    postAction is ArmProviderActionAsync<
```
Comment (timotheeguerin) [inline docs/libraries/azure-resource-manager/rules/arm-resource-invalid-action-verb.md:26]:
```suggestion
    op postAction is ArmProviderActionAsync<
```

**Comment by AlitzelMendez** [comment]
talked with Allen offline. I will close this pr an open a new one with the actual changes for this

**Pull request closed without merging**